### PR TITLE
Run registry DB interface

### PR DESCRIPTION
**Pull request by plasorak**
Interface with the run registry interface:
 - Now creates a `.tgz` of all the config dump and returns its path in `SimpleConfigSaver` (renamed from `ConfigSaver`)
 - Add a dummy `save_on_stop` function in `SimpleConfigSaver`
 - Add run registry credentials, that should be in `~/.nanorc.json`.
 - Add `DBConfigSaver` which saves the `${config_dump}.tgz` created by `SimpleConfigSaver` in the run registry DB
 - User now needs to provide either `--run-type TEST` or `PROD` upon issuing `start` when using `nano04rc` (`nanorc` only uses `SimpleConfigSaver`, and passes `TEST`, but that gets ignored).
 - Stop command is unchanged, but calls `DBConfigSaver.save_on_stop` under the hood.

### CODE REVIEW

**Comment by plasorak** [comment]
https://github.com/DUNE-DAQ/nanorc/wiki for instructions on how to test it

File: src/nanorc/cfgsvr.py [hunk 1]
```diff
@@ -79,7 +96,74 @@ def save_on_resume(self, data: dict) -> dict:
...
 
-
+    
+    def save_on_stop(self, run:int):
+        pass
+        
+
+        
+
+class DBConfigSaver(SimpleConfigSaver):
```
Comment (alessandrothea) [inline src/nanorc/cfgsvr.py:106]: Sorry, I've just noticed this now.
Why `DBConfigSaver` inherits from `SimpleConfigSaver`.
The two share the interface but only the `SimpleConfigSaver` writes to disk.
I would suggest to either drop inheritance or to create a common base class.

As bonus, the `DBConfigSaver` `cfg_outdir` argument will not be needed anylonger.

**Review by alessandrothea — CHANGES_REQUESTED** [review]

File: src/nanorc/__main_np04__.py [hunk 2]
```diff
@@ -58,9 +59,15 @@ def np04cli(ctx, obj, traceback, loglevel, timeout, cfg_dumpdir, dotnanorc, cfg_
...
+        credentials.add_login("runregistrydb",
+                              dotnanorc["runregistrydb"]["user"],
+                              dotnanorc["runregistrydb"]["password"])
         logging.getLogger("cli").info("RunDB socket "+dotnanorc["rundb"]["socket"])
-        rc = NanoRC(obj.console, cfg_dir, cfg_dumpdir,
-                    DBRunNumberManager(dotnanorc["rundb"]["socket"]), timeout)
+        logging.getLogger("cli").info("RunRegistryDB socket "+dotnanorc["runregistrydb"]["socket"])
+        rc = NanoRC(obj.console, cfg_dir,
+                    DBRunNumberManager(dotnanorc["rundb"]["socket"]),
+                    DBConfigSaver("foobar",dotnanorc["runregistrydb"]["socket"]),
```
Comment (alessandrothea) [inline src/nanorc/__main_np04__.py:69]: "foobar"?

File: src/nanorc/cfgsvr.py [hunk 3]
```diff
@@ -79,7 +98,82 @@ def save_on_resume(self, data: dict) -> dict:
...
+        
+
+class DBConfigSaver:
+    def __init__(self, installation_id:str, socket:str):
+        self.API_SOCKET = socket
+        auth = credentials.get_login("runregistrydb")
+        self.API_USER = auth.user
+        self.API_PSWD = auth.password
+        self.timeout = 2
+        self.installation_id = installation_id
```
Comment (alessandrothea) [inline src/nanorc/cfgsvr.py:115]: Dis we say `installation` or was it `apparatus` at the end?

Comment (plasorak) [reply src/nanorc/cfgsvr.py:115]: \facepalm. That should be apparatus.

**Pull request merged**